lib/active_merchant/billing/gateways/airwallex.rb

```diff
@@ -0,0 +1,257 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class AirwallexGateway < Gateway
+      self.test_url = 'https://api-demo.airwallex.com/api/v1'
+      self.live_url = 'https://pci-api.airwallex.com/api/v1'
+
+      # per https://www.airwallex.com/docs/online-payments__overview, cards are accepted in all EU countries
+      self.supported_countries = %w[AT AU BE BG CY CZ DE DK EE GR ES FI FR GB HK HR HU IE IT LT LU LV MT NL PL PT RO SE SG SI SK]
+      self.default_currency = 'AUD'
+      self.supported_cardtypes = %i[visa master]
+
+      self.homepage_url = 'https://airwallex.com/'
+      self.display_name = 'Airwallex'
+
+      ENDPOINTS = {
+        login: '/authentication/login',
+        setup: '/pa/payment_intents/create',
+        sale: '/pa/payment_intents/%{id}/confirm',
+        capture: '/pa/payment_intents/%{id}/capture',
+        refund: '/pa/refunds/create',
+        void: '/pa/payment_intents/%{id}/cancel'
+      }
+
+      def initialize(options = {})
+        requires!(options, :client_id, :client_api_key)
+        @client_id = options[:client_id]
+        @client_api_key = options[:client_api_key]
+        super
+        setup_ids(options) unless options[:request_id] && options[:merchant_order_id]
+        setup_access_token
+      end
+
+      def purchase(money, card, options = {})
+        payment_intent = create_payment_intent(money, @options)
+        post = {
+          'request_id' => update_request_id(@options, 'purchase'),
+          'return_url' => options[:return_url]
+        }
+        add_card(post, card, options)
+        add_payment_method_options(post, card, options)
+
+        commit(:sale, post, payment_intent)
+      end
+
+      def authorize(money, payment, options = {})
+        # authorize is just a purchase w/o an auto capture
```

Comment:
Thank you for these helpful comments.

Reply:
Glad to help! Do you think we should leave them for posterity, or should I clean them up?

Reply:
I think we should leave them. They are succinct and serve to identify some of the gateway's quirks, which might cause a reader to raise an eyebrow otherwise.